solidity/script/avs/DeployAVS.s.sol

```diff
@@ -0,0 +1,134 @@
+// SPDX-License-Identifier: MIT OR Apache-2.0
+pragma solidity >=0.8.0;
+
+import "forge-std/Script.sol";
+
+import {IStrategy} from "../../contracts/interfaces/avs/vendored/IStrategy.sol";
+import {IAVSDirectory} from "../../contracts/interfaces/avs/vendored/IAVSDirectory.sol";
+import {IPaymentCoordinator} from "../../contracts/interfaces/avs/vendored/IPaymentCoordinator.sol";
+import {IDelegationManager} from "../../contracts/interfaces/avs/vendored/IDelegationManager.sol";
+
+import {ProxyAdmin} from "../../contracts/upgrade/ProxyAdmin.sol";
+import {TransparentUpgradeableProxy} from "../../contracts/upgrade/TransparentUpgradeableProxy.sol";
+import {ECDSAStakeRegistry} from "../../contracts/avs/ECDSAStakeRegistry.sol";
+import {Quorum, StrategyParams} from "../../contracts/interfaces/avs/vendored/IECDSAStakeRegistryEventsAndErrors.sol";
+import {HyperlaneServiceManager} from "../../contracts/avs/HyperlaneServiceManager.sol";
+
+import {TestPaymentCoordinator} from "../../contracts/test/avs/TestPaymentCoordinator.sol";
+
+contract DeployAVS is Script {
+    using stdJson for string;
+
+    struct StrategyInfo {
+        string name;
+        address strategy;
+    }
+
+    uint256 deployerPrivateKey;
+
+    ProxyAdmin public proxyAdmin;
+    IAVSDirectory public avsDirectory;
+    IPaymentCoordinator public paymentCoordinator;
+    IDelegationManager public delegationManager;
+
+    Quorum quorum;
+    uint256 thresholdWeight = 6667;
+
+    function _loadEigenlayerAddresses(string memory targetEnv) internal {
+        string memory root = vm.projectRoot();
+        string memory path = string.concat(
+            root,
+            "/script/avs/eigenlayer_addresses.json"
+        );
+        string memory json = vm.readFile(path);
+
+        proxyAdmin = ProxyAdmin(
+            json.readAddress(
+                string(abi.encodePacked(".", targetEnv, ".proxyAdmin"))
+            )
+        );
+        avsDirectory = IAVSDirectory(
+            json.readAddress(
+                string(abi.encodePacked(".", targetEnv, ".avsDirectory"))
+            )
+        );
+        delegationManager = IDelegationManager(
+            json.readAddress(
+                string(abi.encodePacked(".", targetEnv, ".delegationManager"))
+            )
+        );
+        // paymentCoordinator = IPaymentCoordinator(json.readAddress(string(abi.encodePacked(".", targetEnv, ".paymentCoordinator"))));
+        paymentCoordinator = new TestPaymentCoordinator(); // temporary until Eigenlayer deploys the real one
```

Comment:
do we have a way to update this after deployment?

Reply:
Nope, this is what I was asking Steven for.

Reply:
added a method to update it in the ECDSAStrategyBase